tests/Integration/FailedRequestRepositoryTest.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace Tests\Integration;
+
+use Towa\GebruederWeissWooCommerce\Plugin;
+use Towa\GebruederWeissWooCommerce\FailedRequestQueue\FailedRequest;
+use Towa\GebruederWeissWooCommerce\FailedRequestQueue\FailedRequestRepository;
+
+class FailedRequestRepositoryTest extends \WP_UnitTestCase
+{
+    public function setUp(): void
+    {
+        parent::setUp();
+        Plugin::onActivation();
+    }
+
+    public function test_it_can_create_a_failed_request()
+    {
+        global $wpdb;
+
+        $repository = new FailedRequestRepository();
+
+        $request = $repository->create(12, FailedRequest::FAILED_STATUS, 2);
+
+        $numberOfRows = $wpdb->get_var("SELECT count(*) FROM {$wpdb->prefix}gbw_request_retry_queue WHERE order_id = {$request->getOrderId()} AND status = \"{$request->getStatus()}\" AND failed_attempts = {$request->getFailedAttempts()}");
+        $this->assertSame(1, intval($numberOfRows));
+    }
+
+    public function test_it_can_update_a_failed_request()
+    {
+        $repository = new FailedRequestRepository();
+        $request = $repository->create(12, FailedRequest::FAILED_STATUS, 2);
+
+        $request->setStatus(FailedRequest::SUCCESS_STATUS);
+        $request->incrementFailedAttempts();
+        $repository->update($request);
+
+        $this->assertSame(1, $this->getNumberOfFailedRequestsInDB());
```

Comment:
Wouldn't this assertion be true even if you hadn't updated the request in the repository?

Reply:
Yes, that is true. I'll update the test.